toolkits/postgres/arcade_postgres/tools/database_engine.py

```diff
@@ -0,0 +1,90 @@
+import re
+from typing import Any, ClassVar
+from urllib.parse import urlparse
+
+from arcade_tdk.errors import RetryableToolError
+from sqlalchemy import text
+from sqlalchemy.ext.asyncio import AsyncEngine, create_async_engine
+
+MAX_ROWS_RETURNED = 1000
+DEFAULT_ISOLATION_LEVEL = "READ COMMITTED"
+TEST_QUERY = "SELECT 1"
+ERROR_REMAPPING = {
+    re.compile(r"This result object does not return rows"): "Only SELECT queries are allowed.",
+}
```

Comment:
This is fun - this makes the retryable tool errors better AND helps humans understand what's going on too!